### PR TITLE
Replace Triggers with delegate and event for smooth animation.

### PR DESCRIPTION
**Pull request by shawyunz**
Add delegates and events in two ContentViews.
Remove the triggers in xaml.
Add events invoking and animation in class behind.

### CODE REVIEW

File: src/TodoApp/Features/Todos/Pages/TodosPage.xaml.cs [hunk 1]
```diff
@@ -7,6 +8,28 @@ public partial class TodosPage
         public TodosPage()
         {
             InitializeComponent();
+            TodoContentView.OnMenuClicked += Menu_Clicked;
```
Comment (tuyen-vuduc) [inline src/TodoApp/Features/Todos/Pages/TodosPage.xaml.cs]: Would it be better if we wrap the method inside an EventHandler instance? It'll reduce the coupling.

**Comment by tuyen-vuduc** [comment]
@shawyunz It's awesome, thank you very much for your contribution.

**Comment by shawyunz** [comment]
> @shawyunz It's awesome, thank you very much for your contribution.

Considering to merge it?

**Comment by tuyen-vuduc** [comment]
> Considering to merge it?

Yes, could you take a look at my comment above?

**Comment by shawyunz** [comment]
> > Considering to merge it?
> 
> Yes, could you take a look at my comment above?

Emm, sorry, where is that. 
If you mean comment about the code, didn't see your comment in "files changed", you can review there.

File: src/TodoApp/Features/Todos/Pages/TodosPage.xaml.cs (hunk 1 above)
Comment (tuyen-vuduc) [reply src/TodoApp/Features/Todos/Pages/TodosPage.xaml.cs]: @shawyunz Please check my comment here.

**Review by tuyen-vuduc — CHANGES_REQUESTED** [review]

**Comment by shawyunz** [comment]
And community toolkit is another way to do the animation (not tested)
https://learn.microsoft.com/en-us/xamarin/community-toolkit/behaviors/animationbehavior

File: src/TodoApp/Features/Todos/Pages/TodosPage.xaml.cs [hunk 2]
```diff
@@ -8,28 +7,34 @@ public partial class TodosPage
...
+        protected override void OnAppearing()
         {
-            var vm = BindingContext as TodosPageViewModel;
-            vm.SidebarMenuVisible = !vm.SidebarMenuVisible;
+            base.OnAppearing();
 
-            TodoContentView.TranslateTo(App.Current.MainPage.Width * 0.8, 0, 500, Easing.CubicIn);
-            TodoContentView.ScaleTo(0.85, 500, Easing.CubicIn);
-            SideBarView.TranslateTo(0, 0, 500, Easing.CubicIn);
+            TodoContentView.OnMenuClicked += Menu_ClickedEvent;
```
Comment (tuyen-vuduc) [inline src/TodoApp/Features/Todos/Pages/TodosPage.xaml.cs:16]: By creating a new instance of EventHandler to wrap the method, it always uses WeakReference to make things loose coupling. We don't need to add and remove this way.

**Review by tuyen-vuduc — CHANGES_REQUESTED** [review]

**Pull request merged**